### PR TITLE
Initial Commit

### PR DESCRIPTION
**Pull request by naili-m**
- Get Web Page Source (resisting JS redirections to parse indicators at the final page)
- Get Page Screenshot

### CODE REVIEW

File: web-scraper/operations.py.bak [hunk 1]
```diff
@@ -0,0 +1,41 @@
+from connectors.core.connector import get_logger, ConnectorError
```
Comment (cs-pooja-singh) [inline web-scraper/operations.py.bak]: please remove this file

**Comment by cs-pooja-singh** [comment]
how expensive is each screenshot? if this enabled for URL enrichment for every indicator, what is the volume it can scale upto? also, for parallel requests, how can we limit the number of chrome instances being spawned. if the code is run as a separate listener so that the driver init, browser init etc happens only once, will that help improve the performance?

File: web-scraper/operations.py.bak (hunk 1 above)
Comment (naili-m) [reply web-scraper/operations.py.bak]: done

**Comment by naili-m** [comment]
> how expensive is each screenshot? if this enabled for URL enrichment for every indicator, what is the volume it can scale upto? also, for parallel requests, how can we limit the number of chrome instances being spawned. if the code is run as a separate listener so that the driver init, browser init etc happens only once, will that help improve the performance?

The scalability would depend on the available resources eventually for every URL spins up a new process which takes up to 68MB according to the preliminary tests, for this initial release we can keep it for about 10 URLs per minute or so in the connector documentation. for a later version running the browser as a service spawning a tab for each URL would improve response time and resource usage in general and would allow us to implement queuing with async queries to prevent the connector from eating up all hosts resources.

File: web-scraper/operations.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        return webdriver.Chrome(chrome_options=options, executable_path=chromedriver_path, service_args=['--verbose', '--log-path=/tmp/chromedriver.log'])
+    except Exception as exp:
+        logger.exception('Error Initiating WebDriver {}'.format(exp))
+        raise ConnectorError('Error Initiating WebDriver {}'.format(exp))    
+    
+def get_web_page_source(config, params):
+    try:
+        url = url_parser(params)
+        driver = selenium_init()
+        driver.get(url)
```
Comment (cs-pooja-singh) [inline web-scraper/operations.py:56]: driver.set_page_load_timeout(<some_finite_seconds>)

File: web-scraper/operations.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        options = Options()
+        options.binary_location = "/opt/google/chrome/google-chrome"
+        options.add_argument('--headless')
+        options.add_argument('--disable-gpu')
+        options.add_argument('--ignore-certificate-errors')
+        options.add_argument('--disable-dev-shm-usage')
+        options.add_argument('window-size=1024x768')
+        options.add_argument('--no-sandbox')
+
+        return webdriver.Chrome(chrome_options=options, executable_path=chromedriver_path, service_args=['--verbose', '--log-path=/tmp/chromedriver.log'])
```
Comment (cs-pooja-singh) [inline web-scraper/operations.py]: is the logpath overwritten for every launch? or appended? we will need a log rotation if appended

File: web-scraper/playbooks/playbooks.json [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
...otGKX3ieMFs1FOZEqa9hJ4TNuXwN2tPV0MWOcJfsDaj0E8J3iN4ATEiq69JjFThCi31RJxSZKcIfUjpIrQkSePA4Nz6PgnBHVF2T8Hcb21rsuGwBVCZigVo5Qji2a0TUTZHDH4EVpmKG3wPsEoQ5QntDUoM6CNpWt6mr5GOIsOI6Kwh3ELokgRgkU39pzBbxOMxBi60GNpUUogpWTwJU6+JB/vYpIlyt2QJJ7gl7TDira/YXCnKN2SmGuQHxFFK2YmJo4cWf4J2lhS+5zaPoK+RsgXKD9HKksITxHKofVLkqgjiV9SFB1d16DjGhWP0Qg8HrOxINdo7T/jUFK0eJMi3sGFFwh5TrCBzib4IcfaEikc+AFr19umfTSQ2IJ6SHCuQwpFbFK8zQhWoqPt32aMJstTUqNZrUraYY31HcJrgpshbAKqRvfqU4QJTIpD5kVPlkRYF9H2Dk/NsnpGFF2QZDmDWzHYDiUSkIIhWEJoUbInTR1ZeoYIDZECIw+Rqgd1hVQSoS5QMiaNFqAvCHaNkB0OjyCArDFxjDYxQWiEKBh6wdDnKJkz2BOCvN6eHPQOgoRhyHB1iveASglskNLhXMXgegZXEbxBKI2zDiksQtdoHdAuxdoaQYuSOVpk+EFjgySIJW1bIcWYWGXk0QJ8Qm3P2LhHBD8w5R4ySPTuTkae5dzZvce4mKNEzGbTsdyUlO0NTpyh5UBZNpTlMVVTYa1Hq4TBem6qEhsG4ihHoGm6M7wHnRh6WzJsChR7XN9ssM6SJRovDEGm27gpbHC2QwhBkU0o8pRIB2RIkGKECDO0zMG1KFoGH3BOopXBW0Nta4RskHT0w0CqEoI1CBcQDDgGXFAQFL53BFsy+GobkISY4AR2ULggEUhwOQiNdWtWjSU1ASML4pDhMZT+mBt+jh1W5GEfXeSavcWcrIgxMXR1w6ZeU3cbVptTyrokGlLObYV3DuvBMxBFAu80Q2dI0h3GWU439NjBopQg0jnt0GKbfTo7pa4tWndIBZIRShwilaMdXtIPFwiREOtdVDDYfjsoIHG2IijFYBus9YSwFSH6UNL7gB0meNZgL+ntCGckqUwJ3oPYbCfIWRAZwTv6rqT3FULVRDoHq7BOEPwWdnQSlNDgM3q7wQ43xEqgMCihyfSCajih8i8IvuL/AfwxIRQ0yHerAAAAAElFTkSuQmCC",
```
Comment (cs-pooja-singh) [inline web-scraper/playbooks/playbooks.json]: can point to large_icon.png. The framework takes care of correctly updating it during connector install

File: web-scraper/operations.py (hunk 3 above)
Comment (naili-m) [reply web-scraper/operations.py]: logpath is used during dev, release should have that stripped

Comment (cs-pooja-singh) [reply web-scraper/operations.py]: let's remove that before the merge then

File: web-scraper/operations.py (hunk 2 above)
Comment (naili-m) [reply web-scraper/operations.py:56]: done

File: web-scraper/operations.py (hunk 3 above)
Comment (naili-m) [reply web-scraper/operations.py]: done

**Pull request merged**